### PR TITLE
Memleak fixes

### PR DESCRIPTION
**Pull request by khancyr**
The PR fixes some memleak.

The scripting and SITL one are interesting but not critical. Still nice to make analyzer warning disappeared

I am unsure about the other on CAN as they are just before a Panic. Should we cleanup things before a Panic ?

### CODE REVIEW

File: libraries/AP_Notify/SITL_SFML_LED.cpp [hunk 1]
```diff
@@ -136,6 +136,7 @@ void SITL_SFML_LED::update_serial_LEDs()
         }
         w = new sf::RenderWindow(sf::VideoMode(xsize*(serialLED_size+1), ysize*(serialLED_size+1)), "SerialLED");
         if (!w) {
+            delete w;
```
Comment (magicrub) [inline libraries/AP_Notify/SITL_SFML_LED.cpp:139]: What's the point of this one?

File: libraries/AP_Airspeed/AP_Airspeed_UAVCAN.cpp [hunk 2]
```diff
@@ -37,6 +37,7 @@ void AP_Airspeed_UAVCAN::subscribe_msgs(AP_UAVCAN* ap_uavcan)
 
     const int airspeed_listener_res = airspeed_listener->start(AirspeedCb(ap_uavcan, &handle_airspeed));
     if (airspeed_listener_res < 0) {
+        delete airspeed_listener;
```
Comment (tridge) [inline libraries/AP_Airspeed/AP_Airspeed_UAVCAN.cpp:40]: it makes no sense to free memory before we panic. It just makes for fatter firmware with no benefit

File: libraries/AP_Notify/SITL_SFML_LED.cpp (hunk 1 above)
Comment (tridge) [reply libraries/AP_Notify/SITL_SFML_LED.cpp:139]: there is no point to it

File: libraries/AP_Scripting/generator/src/main.c [hunk 3]
```diff
@@ -191,7 +193,8 @@ void error(const int code, const char *message, ...) {
   va_start(args, message);
   vfprintf(stderr, fmt, args);
   va_end(args);
-
+  free(fmt);
```
Comment (tridge) [inline libraries/AP_Scripting/generator/src/main.c:196]: don't free before an exit

**Review by tridge — CHANGES_REQUESTED** [review]
nearly all of these changes just make for larger firmware with no benefit

**Comment by peterbarker** [comment]
@tridge @magicrub I think @khancyr might have been running some tool pointing out memory leaks.  Would be handy to get real results from such a tool - these fake results hide those real results.

Problem is this makes the embedded firmware larger, and also clutters the code with stuff that really doesn't matter.

So... probably just not worth it, @khancyr ?

**Comment by khancyr** [comment]
Yep that is what I thought.
The one for SITL ( that need correction) and the one for generator should go in as   they don't impact STM build.

**Pull request merged**